compiler/rustc_borrowck/src/universal_regions.rs

```diff
@@ -821,6 +821,8 @@ impl<'tcx> UniversalRegionIndices<'tcx> {
     pub fn to_region_vid(&self, r: ty::Region<'tcx>) -> RegionVid {
         if let ty::ReVar(..) = *r {
             r.to_region_vid()
+        } else if let ty::ReError = *r {
+            RegionVid::new(0)
```

Comment:
This should probably be changed before merging :-/

Reply:
I tried chaning `UniversalRegionIndices::build` to either reuse the `re_static` for `ReError` as its second mapping, or creating a new mapping for `re_error` by creating a second lifetime, but either case would blow up with a multitude of ICEs before even getting past building `core`. I know this is wrong, but I would love it if someone from t-types could take that part on :)

Reply:
no such mappings would work afaict, because they would affect the happy path.

imo we could keep this change by making the comment explain that and adding a constant for the `re_static` `RegionVid` (instead of the magic zero).

With `ReError` having an `ErrorGuaranteed` field, all of that should be fine.

Reply:
Done and done